packages/toolkit/src/view/pipeline/view-pipeline/PipelineRunViews/RunComponents.tsx

```diff
@@ -0,0 +1,250 @@
+import * as React from "react";
+import { ColumnDef, PaginationState } from "@tanstack/react-table";
+import { ComponentRun, GeneralRecord, Nullable } from "instill-sdk";
+
+import { Button, DataTable, Dialog } from "@instill-ai/design-system";
+
+import {
+  ModelSectionHeader,
+  RunsTableSortableColHeader,
+  RunStateLabel,
+} from "../../../../components";
+import {
+  convertToSecondsAndMilliseconds,
+  formatDate,
+  InstillStore,
+  useInstillStore,
+  useRouteInfo,
+  useShallow,
+} from "../../../../lib";
+import { usePaginatedPipelineRunComponents } from "../../../../lib/react-query-service/pipeline";
+import { TABLE_PAGE_SIZE } from "../constants";
+
+const selector = (store: InstillStore) => ({
+  accessToken: store.accessToken,
+  enabledQuery: store.enabledQuery,
+});
+
+export const PipelineRunComponents = ({
+  pipelineRunId,
+}: {
+  pipelineRunId: Nullable<string>;
+}) => {
+  const routeInfo = useRouteInfo();
+  const [orderBy, setOrderBy] = React.useState<string>();
```

Comment:
@orangecoloured To use empty value, please use Nullable<string> to make the empty state explicit